app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java

```diff
@@ -475,11 +475,27 @@ else if(resultCode == RESULT_OK){
                 }
                 break;
             }
-            sessionNavigator.startNextSessionStep();
+            startNextSessionStepSafe();
         }
         super.onActivityResult(requestCode, resultCode, intent);
     }
 
+    private void startNextSessionStepSafe() {
+        try {
+            sessionNavigator.startNextSessionStep();
+        } catch (RuntimeException e) {
```

Comment:
Might be worth only catching fixture specific exceptions, but that would require more work. Let me know what you think